.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,54 @@
+name: SDK Benchmark Tests
+
+on:
+  push:
+    branches: [ main ]
+
+jobs:
+  sdk-benchmarks:
+    env:
+      py311: "3.11"
+      RUN_MATRIX_COMBINATION: ${{ matrix.python-version }}-sdk-${{ matrix.os }}
+    runs-on: self-hosted
+    strategy:
+      # Ensures the entire test matrix is run, even if one permutation fails
+      fail-fast: false
+      matrix:
+        python-version: [py311]
+        os: [ubuntu-20.04, windows-2019]
+    steps:
+    - name: Checkout Core Repo @ SHA - ${{ github.sha }}
+      uses: actions/checkout@v2
+    - name: Set up Python ${{ env[matrix.python-version] }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ env[matrix.python-version] }}
+        architecture: 'x64'
+    - name: Install tox
+      run: pip install tox==3.27.1 -U tox-factor
```

Comment:
Unrelated but fyi tox 4 has `-f` supported out of the box. We should probably update all of our test workflows to use that, and you don't need tox-factor

Reply:
Opened #3506